eventsourcing/src/main/java/org/axonframework/eventsourcing/eventstore/EventStorageEngineToStreamableEventSourceAdapter.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2010-2025. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.eventsourcing.eventstore;
+
+import jakarta.annotation.Nonnull;
+import org.axonframework.eventhandling.EventMessage;
+import org.axonframework.eventhandling.processors.streaming.token.TrackingToken;
+import org.axonframework.eventstreaming.StreamableEventSource;
+import org.axonframework.eventstreaming.StreamingCondition;
+import org.axonframework.messaging.MessageStream;
+
+import java.time.Instant;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Adapter class that bridges an {@link EventStorageEngine} to act as a {@link StreamableEventSource}.
+ * <p>
+ * This adapter translates calls on the {@link StreamableEventSource} interface to the corresponding methods in the
+ * provided {@link EventStorageEngine}.
+ * <p>
+ * This allows the {@link EventStorageEngine} to be used as a source for the
+ * {@link org.axonframework.eventhandling.processors.streaming.StreamingEventProcessor}.
+ *
+ * @author Mateusz Nowak
+ * @since 5.0.0
+ */
+public class EventStorageEngineToStreamableEventSourceAdapter implements StreamableEventSource<EventMessage> {
```

Comment:
I'd prefer we call this the `SimpleStreamableEventSource` to make it align with the our other infrastructure components that are a simple mapping from one component to another.

Reply:
I was considering that, and I'm not sure, because the name `simple` doesn't say anything.
Or do you want to be like: "use this one, without thinking too much" :D ? 
But now the name it's more descriptive, and explicitly shows what's happening here.